### PR TITLE
Ar 99 en tant quutilisateur je souhaite pouvoir me connecter creer un compte en utilisant l oauth

### PR DESCRIPTION
**Pull request by KitetsuK**
* **Checklist before requesting a review**
- [x] The commits messages follows our guidelines
- [x] I have performed a self-review of my code
- [ ] I have documented my code
- [ ] Do we need to implement unit tests?

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of changes does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **Describe your changes**

* **Issue ticket number and link**

## Summary by CodeRabbit

- **New Features**
	- Implemented Google OAuth login functionality.
	- Introduced Google user information fetching service.
	- Added a new `GoogleAuth` component for user authentication in the front-end.
	- Created a route for Google login on the server-side.

- **Enhancements**
	- Updated user authentication logic to support OAuth.
	- Enhanced user model to include OAuth login status.

- **Bug Fixes**
	- Fixed user identification in the middleware to search for encrypted email.

- **Refactor**
	- Removed unnecessary `useState` for `code` in Spotify OAuth component.

- **Style**
	- Updated the visual style of the dropdown in the mobile app top bar.

- **Documentation**
	- Added comments and documentation for new Google OAuth services and interfaces.

### CODE REVIEW

File: back/src/services/API/Google/authService.service.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const authenticateUser  = async (code: string, mobile: boolean): Promise<OAuthData> => {
+    if (!process.env.GOOGLE_REDIRECT_URI_WEB || !process.env.GOOGLE_CLIENT_ID || !process.env.GOOGLE_REDIRECT_URI_MOBILE || !process.env.GOOGLE_SERVICE_ID) {
+        console.error("error : Bad env configuration")
+        throw new Error ("Bad env configuration")
+    }
+    const GOOGLE_SERVICE_ID : number = Number(process.env.GOOGLE_SERVICE_ID)
+    const GOOGLE_REDIRECT_URI : string = getRedirectUri(mobile) ?? ''
+
+    try {
+        console.log("code : ", code)
```
Comment (Vivelis) [inline back/src/services/API/Google/authService.service.ts:27]: should remove debug

File: back/src/services/API/Google/authService.service.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            console.error({errorMessage: "An error was caught ", error: data})
+            throw new Error('Failed to authenticate with Google');
+        }
+        const oauthData: OAuthData = {
+            accessToken: data.access_token,
+            refreshToken: "",
+            expiresIn: data.expires_in,
+            serviceId: GOOGLE_SERVICE_ID,
+        }
+        console.log("\x1b[32mUser successfully connected to spotify, access token = \x1b[0m", oauthData.accessToken)
```
Comment (Vivelis) [inline back/src/services/API/Google/authService.service.ts:51]: google not spotify

File: back/src/services/API/Google/getUserEmail.service.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    const googleInfo: GoogleUserInfo = {
+        name: userData.name,
+        email: userData.email,
+        givenName: userData.given_name,
+        familyName: userData.family_name,
+    }
+    return googleInfo
+}
+
+export { getGoogleUserInfo }
```
Comment (Vivelis) [inline back/src/services/API/Google/getUserEmail.service.ts:24]: EOF

File: front/app/login/page.tsx [hunk 4]
```diff
@@ -23,6 +24,7 @@ export default function Login() {
         }
     }, [loginSuccess, router]);
 
+
```
Comment (Vivelis) [inline front/app/login/page.tsx:27]: should be only one empty line

File: front/app/Components/OauthLogin/googleLogin.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        for ( var i = 0; i < length; i++ ) {
+          result += characters.charAt(Math.floor(Math.random() * charactersLength));
+        }
+        return result;
+    }
+
+    const doAsync = async () => {
+        const queryParams = new URLSearchParams(window.location.search);
+        const queryCode: string | null = queryParams.get('code');
+
```
Comment (Vivelis) [inline front/app/Components/OauthLogin/googleLogin.tsx:41]: should check if `state` correspond to the one sent

**Review by Vivelis — CHANGES_REQUESTED** [review]

File: back/src/index.ts [hunk 6]
```diff
@@ -24,7 +24,8 @@ if (isNaN(INTERVAL)) {
     throw new Error('Invalid INTERVAL value. Please check your environment variable.');
 }
 
-db.sequelize.sync({force: false})
+// Initialize the database connection
+db.sequelize.sync({force: true})// {force: true} to drop the tables and recreate them
```
[the remark this answers is not in the document]
Comment (Vivelis) [reply back/src/index.ts]: good job @coderabbitai !

[the remark this answers is not in the document]
Comment (Vivelis) [reply back/src/index.ts]: suck my dick please

[the remark this answers is not in the document]
Comment (Vivelis) [reply back/src/index.ts]: can you date with me ?

[the remark this answers is not in the document]
Comment (Vivelis) [reply back/src/index.ts]: I love you

**Review by Vivelis — APPROVED** [review]

**Review by Beafowl-Pull — APPROVED** [review]

**Review by Vivelis — APPROVED** [review]

**Review by Beafowl-Pull — APPROVED** [review]
RAS

**Pull request merged**